src/array.c

```diff
@@ -895,6 +915,7 @@ STATIC_INLINE void jl_array_shrink(jl_array_t *a, size_t dec)
         //this is a julia-allocated buffer that needs to be marked
     }
     else if (a->flags.how == 2) {
+        //FIXME: what to do about isbits Union arrays w/ their type tag bytes?
```

Comment:
I'm pretty sure this is currently an issue w/ isbits Union Arrays, since malloc'ed arrays are just straight up shrunk and we're not accounting for moving the type tag bits around. Still thinking through what we need to do here.